runconfig/parse.go

```diff
@@ -513,3 +505,31 @@ func ParseDevice(device string) (DeviceMapping, error) {
 	}
 	return deviceMapping, nil
 }
+
+// ParseVolumesIntoBinds parses any volumes passed by the client and moves any
+// volumes which are in Config.Volumes into HostConfig.Binds.
+func ParseVolumesIntoBinds(c *Config, hc *HostConfig) (*Config, *HostConfig, error) {
+	if c != nil && hc != nil {
+		newVolumes := c.Volumes
+		for bind := range newVolumes {
+			var (
```

Comment:
The `if len(mp.Source)...` block? And move that processing into ParseMountSpec? If so, that would significantly change the purpose of what ParseMountSpec does which is operate on a single spec, validating it and parsing it into its constituent components, and is called from more than just here. ParseMountSpec also has no context as to hostConfig or Config - it's purely operating on a single spec and has a succinct common purpose. What am I missing in your comment as I don't understand it otherwise.